modules/gradle-plugin/src/main/kotlin/org/jetbrains/compose/storytale/plugin/AndroidStorytaleExtension.kt

```diff
@@ -0,0 +1,23 @@
+package org.jetbrains.compose.storytale.plugin
+
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.dsl.KotlinAndroidExtension
+
+class AndroidStorytaleExtension(project: Project) : StorytaleExtension(project) {
+    private val kotlinAndroidExtension = run {
+        val androidClass =
+            tryGetClass<KotlinAndroidExtension>(
+                className = "org.jetbrains.kotlin.gradle.dsl.KotlinAndroidExtension",
+            )
+        androidClass?.let { project.extensions.findByType(it) } ?: error("UNEXPECTED")
+    }
+
+    override val targets = listOf(kotlinAndroidExtension.target)
```

Comment:
Couldn't we move this logic into the `StorytaleExtension` so that we don't need to separate plugins for Android-only and multiplatform projects?